validator/sawtooth_validator/execution/scheduler_parallel.py

```diff
@@ -70,34 +67,147 @@ def _tokenize_address(self, address):
             for i in range(0, len(address), self._token_size)
         ]
 
-    def _get(self, address, create=False):
+    def _get_child(self, node, address):
+        try:
+            child = next(
+                child
+                for child in node.children.values()
+                if address.startswith(child.address)
+            )
+
+        except StopIteration:
+            raise AddressNotInTree()
+
+        return child
+
+    def _walk_to_address(self, address):
+        node = self._root
+
+        yield node
+
+        while node.address < address:
```

Comment:
Could you talk about the importance of the lexicographical ordering here? For example 'aaaa' < 'aaab' but 'aaac > 'aaab'.

Reply:
A node's address is always a strict prefix of all the addresses of its children. An equivalent test would be:
```
        while address != node.address and address.startswith(node.address):
            node = self._get_child(node, address)
```

Reply:
I think he meant put it in a comment

Reply:
comment added